### ISSUE TITLE
Speed up tickets.AnchorFile. [phab:D1528]

### ISSUE THREAD

**Issue by kythe** [google/kythe#1943]
_Created by **fromberger** at 2017-04-14 20:10:45_: 

This function is fairly hot in serving benchmarks, as it is heavily used to map
from anchors to their enclosing file tickets. Incorporate the changes from
#1937 to make things run faster.

- Add a benchmark for the tickets.AnchorFile function.

Pre-#1937

```
$ ./bazel-bin/kythe/go/util/schema/tickets/tickets_test \
  -test.bench=. -test.benchtime=5s -test.count=4
BenchmarkAnchorFile-6            2000000              4717 ns/op
BenchmarkAnchorFile-6            2000000              4699 ns/op
BenchmarkAnchorFile-6            2000000              4708 ns/op
BenchmarkAnchorFile-6            2000000              4735 ns/op
```

Mean: 4714 ns/op

BEFORE (kytheuri.Parse)

```
$ ./bazel-bin/kythe/go/util/schema/tickets/tickets_test \
  -test.bench=. -test.benchtime=5s -test.count=4
BenchmarkAnchorFile-6            3000000              2832 ns/op
BenchmarkAnchorFile-6            3000000              2925 ns/op
BenchmarkAnchorFile-6            3000000              2826 ns/op
BenchmarkAnchorFile-6            3000000              2797 ns/op
```

Mean: 2845 ns/op

AFTER (kytheuri.ParseRaw)

```
$ ./bazel-bin/kythe/go/util/schema/tickets/tickets_test \
  -test.bench=. -test.benchtime=5s -test.count=4
BenchmarkAnchorFile-6           10000000              1048 ns/op
BenchmarkAnchorFile-6           10000000              1070 ns/op
BenchmarkAnchorFile-6           10000000              1091 ns/op
BenchmarkAnchorFile-6           10000000              1115 ns/op
```

Mean: 1081 ns/op (4x over original, 2.6x over plain Parse).

**Comment by kythe** [comment]
_**schroederc** wrote at 2017-04-14 20:49:37_:

👍👍/👍👍

**Comment by kythe** [comment]
_**schroederc** accepted this revision at 2017-04-14 20:49:37_

**Comment by kythe** [comment]
_**fromberger** closed via commit 9d5d7b02e8f7d3d0d43dd84e379bce1c8394e6ac at 2017-04-17 12:08:32_
